### DISCUSSION TITLE
Set up offscreen with typescript

### DISCUSSION THREAD

**Discussion by pandasoli** [crxjs/chrome-extension-tools#919] [General]
Hello everyone. Is it possible to use typescript in a offscreen document?

I tested many ways of putting the html or the ts file of the offscreen page inside `src/` so the ts file would be compiled, but I couldn't figure anyway of doing so. And Vite's config and documentation is a mess.

**Comment by Spixz** [comment]
Hi,
You have to update your `vite.config.ts` to include your `offscreen.html` in the dist folder.
```typescript
build: {
    rollupOptions: {
      input: {
        offscreen: "src/offscreen/offscreen.html",
      },
    },
  },
``` 

Then, from the service worker, create the offscreen document with the path of the `offscreen.html` in the dist folder.
```typescript
  await chrome.offscreen.createDocument({
    url: chrome.runtime.getURL("src/offscreen/offscreen.html"),
    reasons: ["BLOBS", "WORKERS"],
    justification: "File modification with FFmpeg et ImageMagick",
  });
```

**Comment by rendomnet** [comment]
@Spixz not working. 
Failed to load resource: net::ERR_FILE_NOT_FOUND offscreen.ts:1

**Reply by Toumash** [reply]
https://github.com/crxjs/chrome-extension-tools/discussions/1060#discussioncomment-14149269
